lib/items/item-component.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import RefHolder from '../models/ref-holder';
+import {Emitter} from 'event-kit';
+import {autobind} from '../helpers';
+
+export default class ItemComponent extends React.Component {
```

Comment:
I think you're right in that inheritance is inherently (heh) not very react-y. Upon some investigation, I think wrapper component but implemented with render prop pattern might be the best way to go, especially since we have quite some existing render prop patterns in the codebase already.